src/components/OverwriteChangesModal/OverwriteChangesModal.tsx

```diff
@@ -0,0 +1,51 @@
+import { Box, Text, VStack } from "@chakra-ui/react"
+import { Button } from "@opengovsg/design-system-react"
+
+import { WarningModal } from "components/WarningModal"
+
+interface OverwriteChangesModalProps {
+  isOpen: boolean
+  onClose: () => void
+  onProceed: () => void
+}
+
+export const OverwriteChangesModal = (
```

Comment:
could i get some insight on why we're doing this instead of just changing `WarningModal`? is this done for convenience due to how much `WarningModal` is used?

Reply:
This is done mainly because the overwrite modal would not be used on solely the EditPage, but also on the other special pages (starting with the EditHomepage). WarningModal is also used in other places as well, which means changing it would not be possible.

Reply:
Ah so to be clear, it's done as a new component for convenience due to the large scope of change for refactoring the `WarningModal`?

I think this is ok then, maybe we might wanna file an issue so we actually fix the root cause (`WarningModal` being trash)

Reply:
Wait sorry I'm a bit confused. This new OverrideChangesModal still uses WarningModal under the hood, so it isn't really a refactoring per se but rather just moving it out of EditPage so that it can be used in other pages.

Reply:
yeap, i just thought that the "ideal" way to do this would be to change `WarningModal` rather than wrapping `WarningModal` which is why i wanted clarity on why this was done